### PR TITLE
Add Intel Fortran compiler options for Windows

### PR DESCRIPTION
**Pull request by degawa**
The format of compiler options for Windows differs from that for Linux and MacOS.
So, warnings appear for every source.
```
ifort: warning #10145: no action performed for file 'declarations,general,usage,interfaces,unused'
ifort: warning #10145: no action performed for file 'f18'
```

To compile properly using Intel compiler for Windows
- added the corresponding options for Windows.
- added `if()-else()-endif()` command to branch the build configuration based on `CMAKE_SYSTEM_NAME`.
---
I tested with the following environment.
- Windows 10 Pro 20H2 64 bit
- Intel Xeon W-2125
- 64GB RAM
- Intel oneAPI classic version 2021.1
- CMake for windows version  3.20.3
- GNU Make for Windows version 3.81 built for i386-pc-mingw32
- fypp version 3.1, Python 3.8.5 (Anaconda distr)

The following command was executed to configure, build, and install.
```
cmake -B build -G "Unix Makefiles" -DCMAKE_Fortran_COMPILER=ifort -DCMAKE_MAXIMUM_RANK=7
cmake --build build
cmake --install build --prefix path/to/my/project/directory
```
Everything worked correctly except for the known failure below.
```
-- Performing Test f18errorstop - Failed
```

The tests were run with
```
cmake --build build --target test
``` 
and passed all tests.
```
100% tests passed, 0 tests failed out of 34

Label Time Summary:
quadruple_precision    =   0.14 sec*proc (2 tests)

Total Test time (real) =  57.17 sec

The following tests did not run:
          1 - always_skip (Skipped)
```

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -33,11 +33,20 @@ if(CMAKE_Fortran_COMPILER_ID STREQUAL GNU)
     add_compile_options(-std=f2008ts)
   endif()
 elseif(CMAKE_Fortran_COMPILER_ID STREQUAL Intel)
-  add_compile_options(-warn declarations,general,usage,interfaces,unused)
-  if(CMAKE_Fortran_COMPILER_VERSION VERSION_LESS 18.0)
-    add_compile_options(-stand f15)
-  else()
-    add_compile_options(-stand f18)
+  if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
```
Comment (awvwgk) [inline CMakeLists.txt:36]: I think we should use [CMAKE_HOST_SYSTEM_NAME](https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_NAME.html#variable:CMAKE_HOST_SYSTEM_NAME) here in case somebody tries to cross-compile stdlib.
```suggestion
  if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows")
```

**Review by awvwgk — APPROVED** [review]
Thanks for sharing. I only have one minor suggestion.

**Comment by degawa** [comment]
Thanks @awvwgk for the significant suggestion.
I've never used `CMAKE_HOST_SYSTEM_NAME` because I always used the same host and target systems.

**Review by everythingfunctional — APPROVED** [review]
LGTM

**Review by ivan-pi — APPROVED** [review]
Looks good to me as well. As long as this passes the Windows build, I think this can be merged.

**Comment by jvdp1** [comment]
@degawa Could you rebase this branch please?

**Comment by awvwgk** [comment]
I discovered something interesting here, a pull-request workflow will actually run on the merge commit with the branch it is targeting, I'm surprised that the workflow is taken from the feature branch rather than the merge commit. This explains at least why the GCC 7/8 workflows on Ubuntu/MacOS and the Windows GCC/MinGW 8.1 workflow are failing from the CMake error which was introduced when disabling those workflows.

Long story short, I think it is safe to just merge this patch with a *squash and merge* or *rebase and merge*.

**Comment by degawa** [comment]
I am sorry for bothering you. 
I'm not familiar with git/Github, and I realize I have misunderstood that the pull request is to send a patch file like a tar archive.

Now I am trying to resolve the problem.

**Comment by degawa** [comment]
I operated my repository to solve the problem, but I did not get a good result.
I'm embarrassed to say that I'm not sure what I should do anymore.
Merging this pull request will not bring good results to the project, so I withdraw this pull request.
I apologize for wasting your time.

**Comment by jvdp1** [comment]
@degawa I am sorry for your issues with git/github. How could we help you? Could potentially one of us (me (but I am not really familiar with Windows), @awvwgk , others used to Windows...?) have a virtual meeting with you?
